modules/tinymce/src/core/test/ts/browser/UndoManagerTest.ts

```diff
@@ -611,4 +611,67 @@ describe('browser.tinymce.core.UndoManagerTest', () => {
 
     assert.equal(changeEventCounter, 0, 'No events should be detected');
   });
+
+  it('TINY-8641: fire change and set editor dirty if the current content and the undoManager history are incoherent', () => {
+    const itTheExpectedChangeEvent = (e, level, lastLevel): boolean => e && e.level === level && e.lastLevel === lastLevel;
+
+    const editor = hook.editor();
+
+    let changeEventCounter = 0;
+    let currentChangeEvent;
+
+    const initialContent = '<p>some inital content</p>';
+    const manualModifiedLevel = 'a modified last level';
+
+    const fireAndCheckEvent = (event, level, lastLevel) => assert.isTrue(
+      itTheExpectedChangeEvent(event, level, lastLevel),
+      'change event should have last level as the current content of the editor and lastLevel as the last level in undoManager'
+    );
```

Comment:
The function doesn't fire the event it asserts properties on it. Also you could instead have separate assertments for the different properties. That way a better error message with a diff would be displayed. Also it seems that all the assertions send in the same content so why not have it hard coded in the function. Suggested code below should obviously also have types.

```suggestion
    const assertChangeEvent = (event) => {
      assert.equal(e.level, manualModifiedLevel, 'Level has not the expected content');
      assert.equal(e.lastLevel, initialLevel, 'Last level has not the expected content');
    };
```

Reply:
done